components/scream/src/physics/p3/p3_functions_ice_cldliq_wet_growth_impl.hpp

```diff
@@ -15,66 +15,65 @@ ::ice_cldliq_wet_growth(const Spack& rho, const Spack& temp, const Spack& pres,
                         const Spack& kap, const Spack& mu, const Spack& sc, const Spack& qv, const Spack& qc_incld,
                         const Spack& qitot_incld, const Spack& nitot_incld, const Spack& qr_incld,
-                        Smask& log_wetgrowth, Spack& qrcol, Spack& qccol, Spack& qwgrth, Spack& nrshdr, Spack& qcshd)
+                        Smask& log_wetgrowth, Spack& qrcol, Spack& qccol, Spack& qwgrth, Spack& nrshdr, Spack& qcshd,
+                        const Smask& context)
 {
+  using physics = scream::physics::Functions<Scalar, Device>;
 
-   using physics = scream::physics::Functions<Scalar, Device>;
+  constexpr Scalar qsmall = C::QSMALL;
+  constexpr Scalar tmelt  = C::Tmelt;
+  constexpr Scalar twopi  = C::Pi*2;
+  constexpr Scalar zero   = C::ZERO;
+  constexpr Scalar one    = C::ONE;
+  constexpr Scalar cpw    = C::CpLiq;
 
-   constexpr Scalar qsmall = C::QSMALL;
-   constexpr Scalar tmelt  = C::Tmelt;
-   constexpr Scalar twopi  = C::Pi*2;
-   constexpr Scalar zero   = C::ZERO;
-   constexpr Scalar one    = C::ONE;
-   constexpr Scalar cpw    = C::CpLiq;
+  const auto t_is_negative = temp < tmelt;
+  const auto qitot_incld_ge_small = qitot_incld >= qsmall;
+  const auto qc_qr_incld_ge_small = (qc_incld + qr_incld) >= sp(1.0e-6);
+  const auto qccol_qrcol_ge_small = (qccol + qrcol) >= sp(1.0e-10);
 
-   const auto t_is_negative = temp < tmelt;
-   const auto qitot_incld_ge_small = qitot_incld >= qsmall;
-   const auto qc_qr_incld_ge_small = (qc_incld + qr_incld) >= sp(1.0e-6);
-   const auto qccol_qrcol_ge_small = (qccol + qrcol) >= sp(1.0e-10);
+  const auto any_if     = qitot_incld_ge_small && qc_qr_incld_ge_small && t_is_negative && context;
+  const auto any_if_col = any_if && qccol_qrcol_ge_small && context;
 
-   const auto any_if     = qitot_incld_ge_small && qc_qr_incld_ge_small && t_is_negative;
-   const auto any_if_col = any_if && qccol_qrcol_ge_small;
+  const Spack zerodeg{tmelt};
 
-   const Spack zerodeg{tmelt};
+  Spack qsat0{0.};
+  Spack dum{0.};
+  Spack dum1{0.};
 
-   Spack qsat0{0.};
-   Spack dum{0.};
-   Spack dum1{0.};
+  if (any_if.any()) {
+    qsat0 = physics::qv_sat( zerodeg,pres,0 );
 
-   if (any_if.any()) {
-     qsat0 = physics::qv_sat( zerodeg,pres,0 );
-     
-      qwgrth.set(any_if,
-                ((f1pr05+f1pr14*pack::cbrt(sc)*sqrt(rhofaci*rho/mu))*
+    qwgrth.set(any_if,
+               ((f1pr05+f1pr14*pack::cbrt(sc)*sqrt(rhofaci*rho/mu))*
                 twopi*(rho*xxlv*dv*(qsat0-qv)-(temp-tmelt)*kap)/
                 (xlf+cpw*(temp-tmelt)))*nitot_incld);
 
-      qwgrth.set(any_if,
-                 pack::max(qwgrth, zero));
+    qwgrth.set(any_if,
+               pack::max(qwgrth, zero));
 
-      dum = pack::max(zero, (qccol+qrcol)-qwgrth);
+    dum = pack::max(zero, (qccol+qrcol)-qwgrth);
 
-      auto const dum_ge_small = dum >= sp(1.0e-10);
+    auto const dum_ge_small = dum >= sp(1.0e-10) && context;
```

Comment:
Minor: `context && dum>=sp(1.0e-10)` might save a Real comparison (short circuit of &&).

That is, assuming `context[i]=false` is more likely than `dum<1e-10`.

Reply:
Since we are dealing with Masks here, I believe the full Mask objects have to be computed before they are passed along to the Mask operators, so I don't thing any short-circuiting is possible, @ambrad may be able to comment here.